WebApp/src/test/java/ev/projects/webapp/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,66 @@
+package ev.projects.webapp.controllers;
+
+import ev.projects.models.User;
+import ev.projects.webapp.WebAppApplication;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest(classes = WebAppApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String baseUrl;
+
+    @Value("${test.username}")
+    private String username;
+    @Value("${test.password}")
+    private String password;
+
+    @BeforeEach
+    public void start() {
+        baseUrl = "http://localhost:" + port + "/api/users/";
+    }
+
+    @Test
+    public void userTest() {
+        User user = new User(username, password);
+        HttpEntity<User> request = new HttpEntity<>(user);
+        assertEquals(HttpStatus.OK, registerUserRequest(request));
+        assertEquals(HttpStatus.CONFLICT, registerUserRequest(request));
```

Comment:
This fails

Reply:
After f9b1db4 still no change. Although with Postman everything works as well as with previous commit. Must be a racing issue.

Reply:
If I split checks to test methods (call the second registration with the same username in the second test) they pass.

Reply:
Solved with 6fd6cf5. The problem was that returning HttpStatus places the status in the response body. Return type should be `ResponseBody<Void>` with return statement like this:
`return new ResponseEntity<>(null, HttpStatus.CONFLICT);`